### DISCUSSION TITLE
Reflection-I

### DISCUSSION THREAD

**Discussion by ghost** [h3rald/min#89] [Features: On Hold]
We should access **parameters**, **returnings** as quotes from in *operator body*:
```
(:n ( (n integer?) (n 0 >) ) &&) 'posint typeclass

( symbol sym-reflection
	(
		(int | string :t) :begin
		t :end
		posint :step
			==>
		quot :yey
		string :haha
	)
	(
		params puts!
		;
		; For example writes on screen:
		; (
		;	('int 'begin)
		;	('int 'end)
		;	('type:posint 'step)
		; )
		;
		
		returns puts!
		;
		; For example writes on screen:
		; (
		;	('quot 'yey)
		;	('string 'haha)
		; )
		;
	)
) ::


```

**Comment by ghost** [comment]
What do you think about?

**Reply by h3rald** [reply]
Well, it would be easy to implement but... can you provide some examples of some practical use cases for it? What would this let you do that you can't do now? 

It's not exactly reflection, you basically want some a well-known parameter to access inputs and outputs from the inside. Something similar to the arguments object in JavaScript... that's useful for more dynamic programming/meta programming, but in a concatenative language you have a lot of ways to write dynamic code with quotations already...

What are you trying to do now? Something related to overloading again?
